iib/web/models.py

```diff
@@ -365,12 +401,35 @@ def from_json(cls, kwargs):
         if current_user.is_authenticated:
             request_kwargs['user'] = current_user
 
-        request_kwargs['type'] = RequestTypeMapping.__members__['add'].value
-
         request = cls(**request_kwargs)
         request.add_state('in_progress', 'The request was initiated')
         return request
 
+    @staticmethod
+    def _handle_request_exclusive_params(request_kwargs):
+        """Handles database operations specific to add/rm requests."""
+
+        # Assume the request is an add request
```

Comment:
My preference is to not assume here. It's possible another request type will be added in the future and we don't want this method to assume `add`.